integration-testing/src/infra/util.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+use crate::infra::subnet::SubnetNode;
+use crate::infra::DEFAULT_MIN_STAKE;
+use anyhow::anyhow;
+use fvm_shared::crypto::signature::SignatureType;
+use ipc_agent::config::json_rpc_methods;
+use ipc_agent::jsonrpc::{JsonRpcClient, JsonRpcClientImpl};
+use ipc_agent::lotus::message::wallet::WalletKeyType;
+use ipc_agent::server::create::{CreateSubnetParams, CreateSubnetResponse};
+use ipc_agent::server::join::JoinSubnetParams;
+use ipc_agent::server::wallet::import::{WalletImportParams, WalletImportResponse};
+use std::process::Command;
+use std::str::FromStr;
+use std::thread::sleep;
+use std::time::Duration;
+
+pub async fn create_subnet(
+    ipc_agent_url: String,
+    from: String,
+    parent: String,
+    name: String,
+    min_validators: u64,
+) -> anyhow::Result<String> {
+    let json_rpc_client = JsonRpcClientImpl::new(ipc_agent_url.parse()?, None);
+
+    let params = CreateSubnetParams {
+        from: Some(from),
+        parent,
+        name,
+        min_validator_stake: DEFAULT_MIN_STAKE,
+        min_validators,
+        bottomup_check_period: 10,
+        topdown_check_period: 10,
+    };
+
+    Ok(json_rpc_client
+        .request::<CreateSubnetResponse>(
+            json_rpc_methods::CREATE_SUBNET,
+            serde_json::to_value(params)?,
+        )
+        .await?
+        .address)
+}
+
+pub async fn join_subnet(
+    ipc_agent_url: String,
+    from: String,
+    subnet: String,
+    collateral: f64,
+    validator_net_addr: String,
+) -> anyhow::Result<()> {
+    let json_rpc_client = JsonRpcClientImpl::new(ipc_agent_url.parse()?, None);
+
+    // The json rpc server will handle directing the request to
+    // the correct parent.
+    let params = JoinSubnetParams {
```

Comment:
Quick idea: this code looks really similar to the one we use in the agent's CLI commands. Should we de-duplicate this code by creating an agent `client` or `sdk` crate that includes these (the same you have `utils`) and that applications or external crates like this one can use?